### DISCUSSION TITLE
Field resolver array

### DISCUSSION THREAD

**Discussion by juliangsibecas** [MichalLytek/type-graphql#851] [Q&A]
Hi! How it's going? 

I'm working with TypeGraphQL and Apollo Federation. I want to resolve an array of ids like this but it's not working:

```
@FieldResolver(() => [FeedItem])
  async items(): Promise<{ __typename: 'Post' | 'Donation'; id: string }[]> {
    const itemsIds: string[] = [
      '784ade6d-0e2f-4375-9f52-e7ee1868a350',
      '71061fd8-cfe6-4fbc-b2eb-3f4ee03cbfe4'
    ]

    return [{ __typename: 'Post', id: itemsIds[0] }, { __typename: 'Donate', id: itemsIds[1] }]
  }
```

If I resolve only one itemId it works. Any clue where it might be failing?

Thank you in advance!

**Comment by MichalLytek** [comment]
TypeGraphQL doesn't support Apollo-style ` __typename` - you need to return instances of the class, like `new FooBar()`.

However I'm still not sure what your issue is about 😛

**Reply by juliangsibecas** [reply]
Sorry if I did not express myself well. I'm using FieldResolver to resolve a field from another service like this: 
```
@FieldResolver(() => User)
  async author(
    @Root() post: Post
  ): Promise<{ __typename: 'User'; id: string }> {
    return { __typename: 'User', id: post.authorId }
  }
```

So this resolver is in the PostResolver and its going to return an User through the UserReference.
```
export const resolveUserReference = async (
  reference: Pick<User, 'id'>
): Promise<User> => {
  return UserEntity.findOne(reference.id)
}

```

The problem is when I try to return an array of Posts it won't resolve anything.


Am I explaining well? Thank you for the quick response!

**Reply by MichalLytek** [reply]
What is `FeedItem`?
